File: Serene/Serene.Script/BasicSamples/GridFilteredByCriteria.cs
using jQueryApi;
using Serene.Northwind;
using Serenity;
using Serenity.Data;

namespace Serene.BasicSamples
{
    public class GridFilteredByCriteria : ProductGrid
    {
        public GridFilteredByCriteria(jQueryObject container)
            : base(container)
        {
        }

        /// <summary>
        /// OnViewSubmit is a method that is called just before 
        /// a List service call is submitted to server side by grid
        /// data source (view).
        /// 
        /// Service call might be cancelled by returning false.
        /// 
        /// Here we use this method to intercept service call
        /// and add a criteria to list request.
        /// </summary>
        protected override bool OnViewSubmit()
        {
            // only continue if base class returns true (didn't cancel request)
            if (!base.OnViewSubmit())
                return false;

            // view object is the data source for grid (SlickRemoteView)
            // this is an EntityGrid so its Params object is a ListRequest
            var request = (ListRequest)view.Params;

            // list request has a Criteria parameter
            // we use " &= " here because otherwise we might clear 
            // filter set by an edit filter dialog if any.

            request.Criteria &=
                new Criteria(ProductRow.Fields.UnitsInStock) > 10 &
                new Criteria(ProductRow.Fields.CategoryName) != "Condiments" &
                new Criteria(ProductRow.Fields.Discontinued) == 0;

            return true;
        }
    }
}

File: Serene/Serene.Web/Modules/BasicSamples/Grids/GridFilteredByCriteria/GridFilteredByCriteria.ts
/// <reference path="../../../Northwind/Product/ProductGrid.ts" />

namespace Serene.BasicSamples {

    @Serenity.Decorators.registerClass()
    export class GridFilteredByCriteria extends Northwind.ProductGrid {

        constructor(container: JQuery) {
            super(container);
        }

        protected onViewSubmit() {
            // only continue if base class returns true (didn't cancel request)
            if (!super.onViewSubmit()) {
                return false;
            }

            // view object is the data source for grid (SlickRemoteView)
            // this is an EntityGrid so its Params object is a ListRequest
            var request = this.view.params;

            // list request has a Criteria parameter
            // we AND criteria here to existing one because 
            // otherwise we might clear filter set by 
            // an edit filter dialog if any.

            request.Criteria = Serenity.Criteria.and(request.Criteria,
                [['UnitsInStock'], '>', 10],
                [['CategoryName'], '!=', 'Condiments'],
                [['Discontinued'], '=', 0]);

            // TypeScript doesn't support operator overloading
            // so we had to use array syntax above to build criteria.

            // Make sure you write
            // [['Field'], '>', 10] (which means field A is greater than 10)
            // not 
            // ['A', '>', 10] (which means string 'A' is greater than 10

            return true;
        }
    }
}